### PR TITLE
[SMALLFIX] Do not show exception on acceptable warning.

### PR DESCRIPTION
**Pull request by calvinjia**
This leads to confusing output when stack traces are printed but there actually isn't a problem. Only adding this for `BlockAlreadyExists` for now.

### CODE REVIEW

File: core/client-internal/src/main/java/alluxio/client/file/FileInStream.java [hunk 1]
```diff
@@ -69,6 +70,9 @@
   /** Constant error message for block ID not cached. */
   private static final String BLOCK_ID_NOT_CACHED =
       "The block with ID {} could not be cached into Alluxio storage.";
+  /** Error message for cache collision. */
+  private static final String BLOCK_ID_EXISTS_SO_NOT_CACHED =
```
Comment (yupeng9) [inline core/client-internal/src/main/java/alluxio/client/file/FileInStream.java]: Add this to `ErrorMessage` enum?

File: core/client-internal/src/main/java/alluxio/client/file/FileInStream.java [hunk 2]
```diff
@@ -69,6 +70,9 @@
...
+      "The block with ID {} has already been stored, canceling the cache request.";
```
Comment (yupeng9) [inline core/client-internal/src/main/java/alluxio/client/file/FileInStream.java]: stored on this machine?

File: core/client-internal/src/main/java/alluxio/client/file/FileInStream.java [hunk 3]
```diff
@@ -258,11 +272,15 @@ private void checkAndAdvanceBlockInStream() throws IOException {
                 mContext.getAluxioBlockStore().getOutStream(currentBlockId, blockSize, address);
           }
         } catch (IOException e) {
-          LOG.warn(BLOCK_ID_NOT_CACHED, currentBlockId, e);
+          if (e.getCause() instanceof BlockAlreadyExistsException) {
```
Comment (yupeng9) [inline core/client-internal/src/main/java/alluxio/client/file/FileInStream.java]: This handling shows up 5 times. Can you extract it into a function?

File: core/client-internal/src/main/java/alluxio/client/file/FileInStream.java [hunk 4]
```diff
@@ -338,11 +356,15 @@ private void seekBlockInStream(long newPos) throws IOException {
...
+            LOG.warn(BLOCK_ID_EXISTS_SO_NOT_CACHED, getCurrentBlockId());
+          } else {
+            LOG.warn(BLOCK_ID_NOT_CACHED, getCurrentBlockId(), e);
+          }
           mShouldCacheCurrentBlock = false;
         } catch (AlluxioException e) {
           LOG.warn(BLOCK_ID_NOT_CACHED, currentBlockId, e);
-          throw new IOException(e);
+          mShouldCacheCurrentBlock = false;
```
Comment (yupeng9) [inline core/client-internal/src/main/java/alluxio/client/file/FileInStream.java]: Should this be in `finally`?

File: core/client-internal/src/main/java/alluxio/client/file/FileInStream.java [hunk 5]
```diff
@@ -258,11 +272,15 @@ private void checkAndAdvanceBlockInStream() throws IOException {
...
```
Comment (yupeng9) [inline core/client-internal/src/main/java/alluxio/client/file/FileInStream.java]: Should this be in `finally`?

File: core/client-internal/src/main/java/alluxio/client/file/FileInStream.java (hunk 4 above)
Comment (calvinjia) [inline core/client-internal/src/main/java/alluxio/client/file/FileInStream.java]: We only want this to be false when an exception is thrown.

File: core/client-internal/src/main/java/alluxio/client/file/FileInStream.java (hunk 1 above)
Comment (calvinjia) [inline core/client-internal/src/main/java/alluxio/client/file/FileInStream.java]: I will refactor the two error messages in this class to `ErrorMessage` in a separate PR to master. (Ticket: https://alluxio.atlassian.net/browse/ALLUXIO-1826)

File: core/client-internal/src/main/java/alluxio/client/file/FileInStream.java (hunk 2 above)
Comment (calvinjia) [inline core/client-internal/src/main/java/alluxio/client/file/FileInStream.java]: Updated clarify, thanks.

File: core/client-internal/src/main/java/alluxio/client/file/FileInStream.java (hunk 3 above)
Comment (calvinjia) [inline core/client-internal/src/main/java/alluxio/client/file/FileInStream.java]: Good call, updated. I kept the setting `mShouldCacheCurrentBlock` outside the function so we do the setting at the same level for all the different branches.

File: core/client-internal/src/main/java/alluxio/client/file/FileInStream.java (hunk 5 above)
Comment (calvinjia) [inline core/client-internal/src/main/java/alluxio/client/file/FileInStream.java]: We only want this to be false when an exception is thrown.

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by yupeng9** [comment]
LGTM. Thanks for the improvement.

**Pull request merged**